### ISSUE TITLE
Regression in handling of class definition inside Enum in Pyright 1.1.378

### ISSUE THREAD

**Issue by saryou** [microsoft/pyright#9169]
## Environment
- **Python Version**: 3.13.0
- **Pyright Version**: 1.1.378 (issue not present in 1.1.377)

## Description
Starting with Pyright 1.1.378, I encountered a change in the behavior of type inference for classes defined inside an `Enum`. In Python 3.13, classes inside an enum that are not explicitly decorated with `member` should not be treated as enum members. This behavior worked as expected in Pyright 1.1.377, but has changed in Pyright 1.1.378.

## Steps to Reproduce
The following minimal code sample reproduces the issue:

```python
from enum import Enum

class Enum(Enum):
    class Class:
        pass

    a = 1

reveal_type(Enum.a)
reveal_type(Enum.Class)
```

### Expected Output (Pyright 1.1.377)
```
Type of "Enum.a" is "Literal[Enum.a]"
Type of "Enum.Class" is "type[Class]"
```

### Actual Output (Pyright 1.1.378)
```
Type of "Enum.a" is "Literal[Enum.a]"
Type of "Enum.Class" is "Literal[Enum.Class]"
```

**Comment by erictraut** [comment]
I think pyright is working correctly.

As you point out, this behavior applies only to Python 3.13 and newer. If you tell pyright to assume a Python version of 3.13, you will see the behavior you're looking for. I'm guessing that you've configured pyright to use 3.12 or earlier. [Here's](https://pyright-play.net/?pythonVersion=3.13&code=GYJw9gtgBApgdgV2gSwgBzCALlAooiAWACgSBjAGwEMBnGqAWQE98kAKViASgC4SoBUSrXoBhanT7FBMqGhEl%2BgqlAC8UAIyLiIGADcYVCgH0sTNDDbNOAOipcSug0dPnL1gjfEiHxIA) what it looks like if you configure it for 3.13.

Also, if you're using Python 3.11 or newer, you can add a `@enum.nonmember` decorator to the class declaration to tell the enum metaclass not to treat it as a member.

```python
from enum import Enum, nonmember

class MyEnum(Enum):
    @nonmember
    class Class:
        pass

    a = 1

reveal_type(MyEnum.a)
reveal_type(MyEnum.Class)
```

**Comment by saryou** [comment]
@erictraut

Thank you for your response.

As of Pyright 1.1.372, the change to infer `Literal[Enum.Class]` for Python versions prior to 3.13 was introduced (https://github.com/microsoft/pyright/pull/8385). I am currently running Python 3.13, and up until Pyright 1.1.377, the output correctly displayed `type[Class]`. If I were running this code with Python 3.12, it would indeed output `Literal[Enum.Class]` as expected, so the behavior change in 1.1.378 seems unusual.

If there's something I might be missing, could you let me know if Pyright's method for determining the Python version being used has changed in 1.1.378?

Thank you again for your time and support.

**Comment by erictraut** [comment]
There was a bug fix in the [1.1.378](https://github.com/microsoft/pyright/releases/tag/1.1.378 release) relating to `pythonVersion`. I'm guessing that you have a `pyproject.toml` configuration file with a `[tools.pyright]` section. Do you have a `pythonVersion` specified there? If you don't specify a `pythonVersion`, pyright will try to use the configured python environment to determine the version. If there is no configured python environment, it defaults to 3.12, which is the most recent stable release.

**Comment by erictraut** [comment]
> Python 3.13.0 was released on 10/7

Thanks for telling me. Looks like it was officially released just a few hours ago. I will bump the default value in pyright to 3.13 for the next release.
